connector/connect/common/src/main/java/org/apache/spark/api/java/function/FlatMapGroupsWithStateFunction.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.api.java.function;
+
+import java.io.Serializable;
+import java.util.Iterator;
+
+import org.apache.spark.annotation.Evolving;
+import org.apache.spark.annotation.Experimental;
+import org.apache.spark.sql.streaming.GroupState;
+
+/**
+ * ::Experimental::
+ * Base interface for a map function used in
+ * {@code org.apache.spark.sql.KeyValueGroupedDataset.flatMapGroupsWithState(
+ * FlatMapGroupsWithStateFunction, org.apache.spark.sql.streaming.OutputMode,
+ * org.apache.spark.sql.Encoder, org.apache.spark.sql.Encoder)}
+ * @since 2.1.1
```

Comment:
nit: version

Reply:
This funcs need to be moved to the util package. I will leave it to you if you want to directly depends on it or copy the files for now.

In the future we should refactoring with other such classes in the util package.

Reply:
The funcs need to be under common package since it's also being used by `UdfUtils`. I don't see a util package in common, do you want me to create one? I think previously you mentioned that there was an effort to build that util package and framework, is that already completed?

Reply:
common currently depends on core and catalyst, so you should have your dependencies there too? In the future, common will depends on some `util` package, which should contains these two classes of yours.

If yours is in core, catalyst, you can leave the refactoring to us later.

Reply:
The dependencies are actually in `sql/core` which is not accessible from the common package, so we might have to duplicate these 2 functions. Let me know which util package I should include these functions, otherwise we can leave them as it is and refactor them later when we have a solid framework there.